src/features/filter.ts

```diff
@@ -1,12 +1,32 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, PayloadAction, Slice } from '@reduxjs/toolkit';
+import { FilterOptions } from '../types';
 
 const initialState = {
   query: '',
-  status: 'all',
+  select: 'all' as FilterOptions,
 };
 
-export const filterSlice = createSlice({
+type FilterState = {
+  query: string;
+  select: FilterOptions;
+};
+
+export const filterSlice: Slice<FilterState> = createSlice({
   name: 'filter',
   initialState,
-  reducers: {},
+  reducers: {
+    setQuery(state, action: PayloadAction<string>) {
+      return {
+        ...state,
+        query: action.payload,
+      };
```

Comment:
By the way,  Redux Toolkit's createSlice uses Immer under the hood, which allows you to mutate the state directly without needing to return a new object.
```suggestion
      state.query = action.payload;
    },
```

Reply:
![image](https://github.com/user-attachments/assets/5ed8839f-9ce7-4ab3-a766-3b2b7afe1e52)
This is good advice, thanks, but i'm breaking eslint rules this way😣

Reply:
Yeah, I know. That's why these rules were added to the mate academy's homeworks)
Rules are made to be broken © 🤣👍🏻